DisguiseUnityRenderStream/ObjectField.cs

```diff
@@ -0,0 +1,112 @@
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Reflection.Emit;
+
+namespace Disguise.RenderStream
+{
+    /// <summary>
+    /// Contains a dynamically-generated collection of methods to set a property or field
+    /// of a given type.
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    /// <remarks>
+    /// This class can be used to avoid boxing operations for calling property or field
+    /// setters when the parameter type is defined statically.
+    /// </remarks>
+    static class DynamicSetterCache<T>
+    {
+        static readonly Dictionary<MemberInfo, Action<object, T>> k_Cache = new();
+
+        public static Action<object, T> GetSetter(MemberInfo member)
+        {
+            var newMethod = new DynamicMethod($"{typeof(T).FullName}.argument_set_{member.Name}",
```

Comment:
Hooo...  Generating code at runtime is "that easy"?  Nice!

Reply:
That just creates the method stub. You still gotta emit IL to make it do stuff 😝